lib/presto/client/statement_client.rb

```diff
@@ -91,22 +90,30 @@ def debug?
       !!@options[:debug]
     end
 
-    def closed?
-      @closed
+    def running?
+      @state == :running
     end
 
-    attr_reader :exception
+    def client_aborted?
+      @state == :client_aborted
+    end
+
+    def client_error?
+      @state == :client_aborted
```

Comment:
Should it be like this?

```suggestion
      @state == :client_error
```

Reply:
good catch!